### ISSUE TITLE
After the upgrade to 2.3.1, the validation accuracy dramatically decreased.

### ISSUE THREAD

**Issue by aerkanc** [keras-team/keras#13723]
I'm using keras with tensorflow backend. My model summary is:
```
Create Model:
========================
Model: "sequential_2"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
embedding_2 (Embedding)      (None, 250, 100)          1847800   
_________________________________________________________________
spatial_dropout1d_2 (Spatial (None, 250, 100)          0         
_________________________________________________________________
gru_3 (GRU)                  (None, 250, 1024)         3456000   
_________________________________________________________________
gru_4 (GRU)                  (None, 256)               983808    
_________________________________________________________________
dense_2 (Dense)              (None, 62)                15934     
=================================================================
Total params: 6,303,542
Trainable params: 6,303,542
Non-trainable params: 0 
```

If i run with keras 2.2.5 version
result is:
```Train on 60015 samples, validate on 6668 samples
Epoch 1/75
60015/60015 [==============================] - 720s 12ms/step - loss: 1.6814 - acc: 0.5311 - val_loss: 1.0556 - val_acc: 0.6738
```

But if i run with keras 2.3.1 version
result is:

```Train on 60015 samples, validate on 6668 samples
Epoch 1/75
60015/60015 [==============================] - 887s 14ms/step - loss: 1.8891 - accuracy: 0.4922 - val_loss: 5.8468 - val_accuracy: 0.0332
```

**Comment by kumariko** [comment]
@aerkanc We are checking to see if you still need help on this issue. We recommend that you upgrade to TF2.6 which is latest stable version of TF and let us know if the issue still persists in newer versions. If it is resolved then please feel free to move this issue to close status.Thanks!
